### ISSUE TITLE
Bug in Firedrake+PETSc Profiling/Logging with firedrake.MG's dg_injection_kernel

### ISSUE THREAD

**Issue by Alexey-Voronin** [firedrakeproject/firedrake#3413]
This issue feels pretty similar to issue #3205, mainly because we're seeing the same kind of problem with the wrong events not being passed to the kernel construction call. I ran into this while playing around with field-split preconditioners for Stokes equations that were discretized with Scott-Vogelius elements. This error pops up even if we don't use barycentric mesh refinement, but just a heads up, without that refinement, the solver's not going to make it to convergence.

**Steps to Reproduce**
```python
from firedrake import *
from firedrake.petsc import PETSc
import petsc4py
petsc4py.PETSc.Sys.popErrorHandler()
PETSc.Log.begin()

class DGMassInv(PCBase):
    def initialize(self, pc):
        _, P = pc.getOperators()
        appctx = self.get_appctx(pc)
        V = dmhooks.get_function_space(pc.getDM())
        # get function spaces
        u = TrialFunction(V)
        v = TestFunction(V)
        massinv = assemble(Tensor(inner(u, v)*dx).inv)
        self.massinv = massinv.petscmat
    def update(self, pc):
        pass
    def apply(self, pc, x, y):
        self.massinv.mult(x, y)
    def applyTranspose(self, pc, x, y):
        raise NotImplementedError("Sorry!")

N = 8
M = UnitSquareMesh(N, N, quadrilateral=False)
M = MeshHierarchy(M,2)[-1]
V = VectorFunctionSpace(M, "CG", 2)
W = FunctionSpace(M, "DG", 1)
Z = V * W

u, p = TrialFunctions(Z)
v, q = TestFunctions(Z)

a = (inner(grad(u), grad(v)) - inner(p, div(v)) + inner(div(u), q))*dx
L = inner(Constant((0, 0)), v) * dx
bcs = [DirichletBC(Z.sub(0), Constant((1, 0)), (4,)),
       DirichletBC(Z.sub(0), Constant((0, 0)), (1, 2, 3))]

up = Function(Z)
nullspace = MixedVectorSpaceBasis(
    Z, [Z.sub(0), VectorSpaceBasis(constant=True)])

solver_parameters = {
 'ksp_error_if_not_converged': False,
 'ksp_max_it': 80,
 'ksp_rtol': 1e-5,
 'ksp_type': 'fgmres',
 'mat_type': 'nest',
 'pc_fieldsplit_schur_factorization_type': 'full',
 'pc_fieldsplit_type': 'schur',
 'pc_type': 'fieldsplit',
 'fieldsplit_0': {'ksp_type': 'preonly',
                  'mat_type': 'aij',
                  'mg_coarse_assembled': {'mat_type': 'aij',
                                          'pc_factor_mat_solver_type': 'mumps',
                                          'pc_type': 'lu'},
                  'mg_coarse_pc_python_type': 'firedrake.AssembledPC',
                  'mg_coarse_pc_type': 'python',
                  'mg_levels': {'ksp_chebyshev_esteig': '0,0.25,0,1.1',
                                'ksp_convergence_test': 'skip',
                                'ksp_max_it': '2',
                                'ksp_norm_type': 'unpreconditioned',
                                'ksp_type': 'chebyshev',
                                'patch_pc_patch_construct_dim': 0,
                                'patch_pc_patch_construct_type': 'star',
                                'patch_pc_patch_local_type': 'additive',
                                'patch_pc_patch_partition_of_unity': False,
                                #'patch_pc_patch_precompute_element_tensors': True,
                                #'patch_pc_patch_save_operators': True,
                                #'patch_pc_patch_statistics': False,
                                'patch_pc_patch_sub_mat_type': 'seqaij',
                                'patch_pc_patch_symmetrise_sweep': False,
                                'patch_sub_ksp_type': 'preonly',
                                'patch_sub_pc_factor_mat_solver_type': 'petsc',
                                'patch_sub_pc_type': 'lu',
                                'pc_python_type': 'firedrake.PatchPC',
                                'pc_type': 'python'},
                  'pc_cycle_type': 'V',
                  'pc_mg_type': 'multiplicative',
                  'pc_type': 'mg'},
 'fieldsplit_1': {'ksp_type': 'preonly',
                  'pc_python_type': '__main__.DGMassInv',
                  'pc_type': 'python'},
 }

with PETSc.Log.Stage(f"solve"):
    solve(a == L, up, bcs=bcs, nullspace=nullspace,
          solver_parameters=solver_parameters
          )
PETSc.Log.view()
```

**Expected behavior**
We shouldn't be seeing any PETSc error messages popping up. Given that we're working with non-barycentrically refined triangular grids, it's totally expected that the current solver won't reach convergence.

**Error message**
The repeated error message looks like this 
```
[0]PETSC ERROR: --------------------- Error Message --------------------------------------------------------------
[0]PETSC ERROR:   It appears a new error in the code was triggered after a previous error, possibly because:
[0]PETSC ERROR:   -  The first error was not properly handled via (for example) the use of
[0]PETSC ERROR:      PetscCall(TheFunctionThatErrors()); or
[0]PETSC ERROR:   -  The second error was triggered while handling the first error.
[0]PETSC ERROR:   Above is the traceback for the previous unhandled error, below the traceback for the next error
[0]PETSC ERROR:   ALL ERRORS in the PETSc libraries are fatal, you should add the appropriate error checking to the code
[0]PETSC ERROR: Argument out of range
[0]PETSC ERROR: Index -1 is not in range [0,562)
[0]PETSC ERROR: WARNING! There are unused option(s) set! Could be the program crashed before usage or a spelling mistake, etc!
[0]PETSC ERROR:   Option left: name:-firedrake_0_fieldsplit_0_mg_coarse_assembled_mat_type value: aij source: code
[0]PETSC ERROR:   Option left: name:-firedrake_0_fieldsplit_0_mg_coarse_assembled_pc_factor_mat_solver_type value: mumps source: code
[0]PETSC ERROR:   Option left: name:-firedrake_0_fieldsplit_0_mg_coarse_assembled_pc_type value: lu source: code
[0]PETSC ERROR:   Option left: name:-firedrake_0_fieldsplit_0_mg_coarse_pc_python_type value: firedrake.AssembledPC source: code
[0]PETSC ERROR:   Option left: name:-firedrake_0_fieldsplit_0_mg_coarse_pc_type value: python source: code
[0]PETSC ERROR:   Option left: name:-firedrake_0_fieldsplit_0_mg_levels_ksp_chebyshev_esteig value: 0,0.25,0,1.1 source: code
[0]PETSC ERROR:   Option left: name:-firedrake_0_fieldsplit_0_mg_levels_ksp_convergence_test value: skip source: code
[0]PETSC ERROR:   Option left: name:-firedrake_0_fieldsplit_0_mg_levels_ksp_max_it value: 2 source: code
[0]PETSC ERROR:   Option left: name:-firedrake_0_fieldsplit_0_mg_levels_ksp_norm_type value: unpreconditioned source: code
[0]PETSC ERROR:   Option left: name:-firedrake_0_fieldsplit_0_mg_levels_ksp_type value: chebyshev source: code
[0]PETSC ERROR:   Option left: name:-firedrake_0_fieldsplit_0_mg_levels_patch_pc_patch_construct_dim value: 0 source: code
[0]PETSC ERROR:   Option left: name:-firedrake_0_fieldsplit_0_mg_levels_patch_pc_patch_construct_type value: star source: code
[0]PETSC ERROR:   Option left: name:-firedrake_0_fieldsplit_0_mg_levels_patch_pc_patch_local_type value: additive source: code
[0]PETSC ERROR:   Option left: name:-firedrake_0_fieldsplit_0_mg_levels_patch_pc_patch_partition_of_unity value: false source: code
[0]PETSC ERROR:   Option left: name:-firedrake_0_fieldsplit_0_mg_levels_patch_pc_patch_sub_mat_type value: seqaij source: code
[0]PETSC ERROR:   Option left: name:-firedrake_0_fieldsplit_0_mg_levels_patch_pc_patch_symmetrise_sweep value: false source: code
[0]PETSC ERROR:   Option left: name:-firedrake_0_fieldsplit_0_mg_levels_patch_sub_ksp_type value: preonly source: code
[0]PETSC ERROR:   Option left: name:-firedrake_0_fieldsplit_0_mg_levels_patch_sub_pc_factor_mat_solver_type value: petsc source: code
[0]PETSC ERROR:   Option left: name:-firedrake_0_fieldsplit_0_mg_levels_patch_sub_pc_type value: lu source: code
[0]PETSC ERROR:   Option left: name:-firedrake_0_fieldsplit_0_mg_levels_pc_python_type value: firedrake.PatchPC source: code
[0]PETSC ERROR:   Option left: name:-firedrake_0_fieldsplit_0_mg_levels_pc_type value: python source: code
[0]PETSC ERROR:   Option left: name:-firedrake_0_fieldsplit_0_pc_cycle_type value: V source: code
[0]PETSC ERROR:   Option left: name:-firedrake_0_ksp_error_if_not_converged value: false source: code
[0]PETSC ERROR:   Option left: name:-firedrake_0_ksp_max_it value: 80 source: code
[0]PETSC ERROR:   Option left: name:-firedrake_0_ksp_rtol value: 1e-05 source: code
[0]PETSC ERROR:   Option left: name:-firedrake_0_ksp_type value: fgmres source: code
[0]PETSC ERROR:   Option left: name:-firedrake_0_mat_mumps_icntl_14 value: 200 source: code
[0]PETSC ERROR:   Option left: name:-firedrake_0_mat_type value: nest source: code
[0]PETSC ERROR:   Option left: name:-firedrake_0_pc_fieldsplit_schur_factorization_type value: full source: code
[0]PETSC ERROR:   Option left: name:-firedrake_0_pc_fieldsplit_type value: schur source: code
[0]PETSC ERROR:   Option left: name:-firedrake_0_pc_type value: fieldsplit source: code
[0]PETSC ERROR:   Option left: name:-firedrake_0_snes_type value: ksponly source: code
[0]PETSC ERROR: See https://petsc.org/release/faq/ for trouble shooting.
[0]PETSC ERROR: Petsc Development GIT revision: v3.4.2-43642-g09f36907a6  GIT Date: 2023-12-08 19:37:01 +0000
[0]PETSC ERROR: stokes2.py on a default named koelsch by voronin2 Sat Feb 17 14:21:11 2024
[0]PETSC ERROR: Configure options PETSC_DIR=/home/voronin2/firedrake/src/petsc PETSC_ARCH=default --download-chaco --with-zlib --with-fortran-bindings=0 --download-pastix --download-hypre --download-pnetcdf --download-suitesparse --download-mpich --download-hdf5 --download-cmake --download-scalapack --with-shared-libraries=1 --download-mumps --download-superlu_dist --download-hwloc --with-debugging=0 --download-ptscotch --download-bison --with-c2html=0 --download-netcdf --download-metis
[0]PETSC ERROR: #1 PetscLogEventPerfArrayGetRef() at /home/voronin2/firedrake/src/petsc/src/sys/logging/handler/impls/default/logdefault.c:159
[0]PETSC ERROR: #2 PetscLogHandlerGetEventPerfInfo_Default() at /home/voronin2/firedrake/src/petsc/src/sys/logging/handler/impls/default/logdefault.c:343
[0]PETSC ERROR: #3 PetscLogHandlerEventBegin_Default() at /home/voronin2/firedrake/src/petsc/src/sys/logging/handler/impls/default/logdefault.c:549
```

Backtrace - 
```
#0  PetscLogHandlerEventEnd_Default (h=0x3206980, event=-1, o1=<optimized out>, o2=<optimized out>, o3=<optimized out>, o4=<optimized out>)
    at /home/voronin2/firedrake/src/petsc/src/sys/logging/handler/impls/default/logdefault.c:614
#1  0x00007fffcdc56814 in PetscLogEventEnd_Internal (o1=<optimized out>, o2=<optimized out>, o3=<optimized out>, o4=<optimized out>, e=<optimized out>)
    at /home/voronin2/firedrake/src/petsc/include/petsclog.h:360
#2  slate_wrapper (w_0=<synthetic pointer>, coords=<synthetic pointer>, output=<synthetic pointer>)
    at /home/voronin2/firedrake/.cache/pyop2/46/7418a7117711ea6b5887f6e932e762_p3797712.c:172
#3  pyop2_kernel_injection_dg (coords=<optimized out>, macro_coords=<optimized out>, macro_w_0=<optimized out>, R=<optimized out>)
    at /home/voronin2/firedrake/.cache/pyop2/46/7418a7117711ea6b5887f6e932e762_p3797712.c:141
#4  wrap_pyop2_kernel_injection_dg (start=<optimized out>, end=<optimized out>, dat0=0x3a9b590, dat1=<optimized out>, dat2=0x33f80c0, dat3=<optimized out>, map0=0x3a0a500, 
    map1=0x3b44c50, map2=0x3b4ac60, map3=0x3361220) at /home/voronin2/firedrake/.cache/pyop2/46/7418a7117711ea6b5887f6e932e762_p3797712.c:248
#5  0x00007ffff7b0d40e in ?? () from /lib/x86_64-linux-gnu/libffi.so.8
#6  0x00007ffff7b0c71d in ?? () from /lib/x86_64-linux-gnu/libffi.so.8
#7  0x00007ffff7b0cef3 in ffi_call () from /lib/x86_64-linux-gnu/libffi.so.8
#8  0x00007ffff3aa4867 in _call_function_pointer (argtypecount=<optimized out>, argcount=10, resmem=0x7fffffffae80, restype=<optimized out>, atypes=<optimized out>, 
    avalues=<optimized out>, pProc=0x7fffcdc55780 <wrap_pyop2_kernel_injection_dg>, flags=<optimized out>) at ./Modules/_ctypes/callproc.c:923
#9  _ctypes_callproc (pProc=<optimized out>, argtuple=<optimized out>, flags=<optimized out>, argtypes=<optimized out>, restype=<optimized out>, checker=<optimized out>)
    at ./Modules/_ctypes/callproc.c:1262
#10 0x00007ffff3aa3582 in PyCFuncPtr_call (self=<optimized out>, inargs=<optimized out>, kwds=0x0) at ./Modules/_ctypes/_ctypes.c:4201
#11 0x00000000005637fe in PyObject_Call ()
#12 0x000000000052c731 in _PyEval_EvalFrameDefault ()
#13 0x000000000051c3d7 in _PyObject_FastCallDictTstate ()
#14 0x000000000055ee71 in _PyObject_Call_Prepend ()
#15 0x0000000000653616 in ?? ()
#16 0x00000000005637fe in PyObject_Call ()
#17 0x000000000052c731 in _PyEval_EvalFrameDefault ()
#18 0x00000000005567e5 in _PyFunction_Vectorcall ()
#19 0x00007ffff70c988f in __Pyx_PyObject_Call (func=0x7fffce3fc360, arg=0x7fffbf7635e0, kw=0x7fffbf781dc0) at src/petsc4py/PETSc.c:542543
#20 0x00007ffff728dddb in __pyx_pf_8petsc4py_5PETSc_3Log_14EventDecorator_9decorator_wrapped_func (__pyx_v_kwargs=0x7fffbf741a80, __pyx_v_args=0x7fffbf7635e0, 
    __pyx_self=<optimized out>) at src/petsc4py/PETSc.c:105331
#21 __pyx_pw_8petsc4py_5PETSc_3Log_14EventDecorator_9decorator_1wrapped_func (__pyx_self=<optimized out>, __pyx_args=0x7fffbf7635e0, __pyx_kwds=<optimized out>)
    at src/petsc4py/PETSc.c:39609
#22 0x000000000051490b in _PyObject_MakeTpCall ()
#23 0x000000000055ee01 in _PyObject_Call_Prepend ()
#24 0x0000000000653616 in ?? ()
#25 0x000000000051490b in _PyObject_MakeTpCall ()
#26 0x0000000000528559 in _PyEval_EvalFrameDefault ()
#27 0x00000000005567e5 in _PyFunction_Vectorcall ()
#28 0x000000000052c731 in _PyEval_EvalFrameDefault ()
#29 0x00000000005567e5 in _PyFunction_Vectorcall ()
#30 0x00007ffff70c988f in __Pyx_PyObject_Call (func=0x7fffcbf5be20, arg=0x7fffbfce3b40, kw=0x7fffca592300) at src/petsc4py/PETSc.c:542543
#31 0x00007ffff728dddb in __pyx_pf_8petsc4py_5PETSc_3Log_14EventDecorator_9decorator_wrapped_func (__pyx_v_kwargs=0x7fffbf7b6900, __pyx_v_args=0x7fffbfce3b40, 
    __pyx_self=<optimized out>) at src/petsc4py/PETSc.c:105331
#32 __pyx_pw_8petsc4py_5PETSc_3Log_14EventDecorator_9decorator_1wrapped_func (__pyx_self=<optimized out>, __pyx_args=0x7fffbfce3b40, __pyx_kwds=<optimized out>)
    at src/petsc4py/PETSc.c:39609
#33 0x000000000051490b in _PyObject_MakeTpCall ()
#34 0x0000000000528559 in _PyEval_EvalFrameDefault ()
#35 0x00000000005567e5 in _PyFunction_Vectorcall ()
#36 0x00000000006874cc in PyVectorcall_Call ()
#37 0x00007ffff70c988f in __Pyx_PyObject_Call (func=0x7fffcbf68ae0, arg=0x7fffbfcc9800, kw=0x7fffbf7a7540) at src/petsc4py/PETSc.c:542543
#38 0x00007ffff728dddb in __pyx_pf_8petsc4py_5PETSc_3Log_14EventDecorator_9decorator_wrapped_func (__pyx_v_kwargs=0x7fffbf7a7480, __pyx_v_args=0x7fffbfcc9800, 
    __pyx_self=<optimized out>) at src/petsc4py/PETSc.c:105331
--Type <RET> for more, q to quit, c to continue without paging-- 
#39 __pyx_pw_8petsc4py_5PETSc_3Log_14EventDecorator_9decorator_1wrapped_func (__pyx_self=<optimized out>, __pyx_args=0x7fffbfcc9800, __pyx_kwds=<optimized out>)
    at src/petsc4py/PETSc.c:39609
#40 0x000000000051490b in _PyObject_MakeTpCall ()
#41 0x0000000000528559 in _PyEval_EvalFrameDefault ()
#42 0x00000000005567e5 in _PyFunction_Vectorcall ()
#43 0x0000000000563868 in PyObject_Call ()
#44 0x000000000052c731 in _PyEval_EvalFrameDefault ()
#45 0x00000000005567e5 in _PyFunction_Vectorcall ()
#46 0x0000000000563868 in PyObject_Call ()
#47 0x000000000052c731 in _PyEval_EvalFrameDefault ()
#48 0x00000000005567e5 in _PyFunction_Vectorcall ()
#49 0x000000000052c731 in _PyEval_EvalFrameDefault ()
#50 0x00000000005567e5 in _PyFunction_Vectorcall ()
#51 0x00007ffff70c988f in __Pyx_PyObject_Call (func=0x7fffce3305e0, arg=0x7fffbfccee40, kw=0x7fffbfccf580) at src/petsc4py/PETSc.c:542543
#52 0x00007ffff728dddb in __pyx_pf_8petsc4py_5PETSc_3Log_14EventDecorator_9decorator_wrapped_func (__pyx_v_kwargs=0x7fffbfccf240, __pyx_v_args=0x7fffbfccee40, 
    __pyx_self=<optimized out>) at src/petsc4py/PETSc.c:105331
#53 __pyx_pw_8petsc4py_5PETSc_3Log_14EventDecorator_9decorator_1wrapped_func (__pyx_self=<optimized out>, __pyx_args=0x7fffbfccee40, __pyx_kwds=<optimized out>)
    at src/petsc4py/PETSc.c:39609
#54 0x00007ffff70c988f in __Pyx_PyObject_Call (func=0x7fffce2ea400, arg=0x7fffbfccee40, kw=0x7fffca585540) at src/petsc4py/PETSc.c:542543
#55 0x00007ffff711ebdf in __pyx_f_8petsc4py_5PETSc_DMSHELL_Coarsen (__pyx_v_dm=<optimized out>, __pyx_v_comm=<optimized out>, __pyx_v_dmc=0x3a50858) at src/petsc4py/PETSc.c:90565
#56 0x00007ffff5d7a7df in DMCoarsen (dm=0x382d920, comm=comm@entry=67108864, dmc=0x3a50858) at /home/voronin2/firedrake/src/petsc/src/dm/interface/dm.c:3291
#57 0x00007ffff637cbbc in PCSetUp_MG (pc=<optimized out>) at /home/voronin2/firedrake/src/petsc/src/ksp/pc/impls/mg/mg.c:961
#58 0x00007ffff6424b2c in PCSetUp (pc=0x316c250) at /home/voronin2/firedrake/src/petsc/src/ksp/pc/interface/precon.c:1079
#59 0x00007ffff622641b in KSPSetUp (ksp=ksp@entry=0x3903ab0) at /home/voronin2/firedrake/src/petsc/src/ksp/ksp/interface/itfunc.c:415
#60 0x00007ffff6228358 in KSPSolve_Private (ksp=ksp@entry=0x3903ab0, b=0x3a15ba0, x=<optimized out>) at /home/voronin2/firedrake/src/petsc/src/ksp/ksp/interface/itfunc.c:836
#61 0x00007ffff622b833 in KSPSolve (ksp=ksp@entry=0x3903ab0, b=<optimized out>, x=<optimized out>) at /home/voronin2/firedrake/src/petsc/src/ksp/ksp/interface/itfunc.c:1083
#62 0x00007ffff6249936 in PCApply_FieldSplit_Schur (pc=0x3241a30, x=<optimized out>, y=0x38e9ad0) at /home/voronin2/firedrake/src/petsc/src/ksp/pc/impls/fieldsplit/fieldsplit.c:1179
#63 0x00007ffff6424fae in PCApply (pc=0x3241a30, x=0x38f7570, y=y@entry=0x38e9ad0) at /home/voronin2/firedrake/src/petsc/src/ksp/pc/interface/precon.c:497
#64 0x00007ffff61bcd0d in KSP_PCApply (y=0x38e9ad0, x=<optimized out>, ksp=0x3239230) at /home/voronin2/firedrake/src/petsc/include/petsc/private/kspimpl.h:383
#65 KSPFGMRESCycle (itcount=itcount@entry=0x7fffffffcef8, ksp=ksp@entry=0x3239230) at /home/voronin2/firedrake/src/petsc/src/ksp/ksp/impls/gmres/fgmres/fgmres.c:123
#66 0x00007ffff61bdb7b in KSPSolve_FGMRES (ksp=0x3239230) at /home/voronin2/firedrake/src/petsc/src/ksp/ksp/impls/gmres/fgmres/fgmres.c:233
#67 0x00007ffff6228f34 in KSPSolve_Private (ksp=0x3239230, b=b@entry=0x322c600, x=<optimized out>, x@entry=0x3832c90)
    at /home/voronin2/firedrake/src/petsc/src/ksp/ksp/interface/itfunc.c:910
#68 0x00007ffff622b833 in KSPSolve (ksp=<optimized out>, b=b@entry=0x322c600, x=x@entry=0x3832c90) at /home/voronin2/firedrake/src/petsc/src/ksp/ksp/interface/itfunc.c:1083
#69 0x00007ffff646eb35 in SNESSolve_KSPONLY (snes=0x3240aa0) at /home/voronin2/firedrake/src/petsc/src/snes/impls/ksponly/ksponly.c:49
#70 0x00007ffff644089b in SNESSolve (snes=0x3240aa0, b=0x0, x=<optimized out>) at /home/voronin2/firedrake/src/petsc/src/snes/interface/snes.c:4659
#71 0x00007ffff716b028 in __pyx_pf_8petsc4py_5PETSc_4SNES_156solve (__pyx_v_self=0x7fffca5c2c50, __pyx_v_x=<optimized out>, __pyx_v_b=<optimized out>) at src/petsc4py/PETSc.c:307670
#72 __pyx_pw_8petsc4py_5PETSc_4SNES_157solve (__pyx_v_self=0x7fffca5c2c50, __pyx_args=<optimized out>, __pyx_nargs=<optimized out>, __pyx_kwds=<optimized out>)
    at src/petsc4py/PETSc.c:45445
#73 0x000000000057201b in ?? ()
#74 0x0000000000536950 in PyObject_Vectorcall ()
#75 0x0000000000528559 in _PyEval_EvalFrameDefault ()
#76 0x00000000005567e5 in _PyFunction_Vectorcall ()
#77 0x000000000052c731 in _PyEval_EvalFrameDefault ()
#78 0x00000000005567e5 in _PyFunction_Vectorcall ()
#79 0x00007ffff70c988f in __Pyx_PyObject_Call (func=0x7fffcdfb0400, arg=0x7fffca6f66e0, kw=0x7fffca5da300) at src/petsc4py/PETSc.c:542543
#80 0x00007ffff728dddb in __pyx_pf_8petsc4py_5PETSc_3Log_14EventDecorator_9decorator_wrapped_func (__pyx_v_kwargs=0x7fffca597ec0, __pyx_v_args=0x7fffca6f66e0, 
    __pyx_self=<optimized out>) at src/petsc4py/PETSc.c:105331
#81 __pyx_pw_8petsc4py_5PETSc_3Log_14EventDecorator_9decorator_1wrapped_func (__pyx_self=<optimized out>, __pyx_args=0x7fffca6f66e0, __pyx_kwds=<optimized out>)
    at src/petsc4py/PETSc.c:39609
#82 0x000000000051490b in _PyObject_MakeTpCall ()
--Type <RET> for more, q to quit, c to continue without paging--
#83 0x0000000000528559 in _PyEval_EvalFrameDefault ()
#84 0x00000000005567e5 in _PyFunction_Vectorcall ()
#85 0x0000000000563868 in PyObject_Call ()
#86 0x000000000052c731 in _PyEval_EvalFrameDefault ()
#87 0x00000000005567e5 in _PyFunction_Vectorcall ()
#88 0x0000000000563868 in PyObject_Call ()
#89 0x000000000052c731 in _PyEval_EvalFrameDefault ()
#90 0x00000000005567e5 in _PyFunction_Vectorcall ()
#91 0x00000000006874cc in PyVectorcall_Call ()
#92 0x00007ffff70c988f in __Pyx_PyObject_Call (func=0x7fffce232700, arg=0x7fffca5860c0, kw=0x7fffca597f00) at src/petsc4py/PETSc.c:542543
#93 0x00007ffff728dddb in __pyx_pf_8petsc4py_5PETSc_3Log_14EventDecorator_9decorator_wrapped_func (__pyx_v_kwargs=0x7fffca5d8400, __pyx_v_args=0x7fffca5860c0, 
    __pyx_self=<optimized out>) at src/petsc4py/PETSc.c:105331
#94 __pyx_pw_8petsc4py_5PETSc_3Log_14EventDecorator_9decorator_1wrapped_func (__pyx_self=<optimized out>, __pyx_args=0x7fffca5860c0, __pyx_kwds=<optimized out>)
    at src/petsc4py/PETSc.c:39609
#95 0x000000000051490b in _PyObject_MakeTpCall ()
#96 0x0000000000528559 in _PyEval_EvalFrameDefault ()
#97 0x00000000006046ff in PyEval_EvalCode ()
#98 0x000000000061e85b in ?? ()
#99 0x000000000061af33 in ?? ()
#100 0x000000000062d666 in ?? ()
#101 0x000000000062d3d1 in _PyRun_SimpleFileObject ()
#102 0x000000000062d1e8 in _PyRun_AnyFileObject ()
#103 0x000000000062b6bb in Py_RunMain ()
#104 0x00000000005f7cab in Py_BytesMain ()
#105 0x00007ffff7cc218a in __libc_start_call_main (main=main@entry=0x5f7c10, argc=argc@entry=2, argv=argv@entry=0x7fffffffe728) at ../sysdeps/nptl/libc_start_call_main.h:58
#106 0x00007ffff7cc2245 in __libc_start_main_impl (main=0x5f7c10, argc=2, argv=0x7fffffffe728, init=<optimized out>, fini=<optimized out>, rtld_fini=<optimized out>, 
    stack_end=0x7fffffffe718) at ../csu/libc-start.c:381
#107 0x00000000005f7b41 in _start ()
```

Add error message with full backtrace, or log.
Please add these as text using three backticks (`) for highlighting.
Please do not add screenshots unless the bug is purely graphical.

**Environment:**
 - OS: Linux comp 6.1.0-5-amd64 1 SMP PREEMPT_DYNAMIC Debian 6.1.12-1 (2023-02-15) x86_64 GNU/Linux
 - Python version:  3.11.7
 - Output of `firedrake-status`
```
Firedrake Configuration:
    package_manager: True
    minimal_petsc: False
    mpicc: None
    mpicxx: None
    mpif90: None
    mpiexec: None
    disable_ssh: False
    honour_petsc_dir: False
    with_parmetis: False
    slepc: False
    packages: []
    honour_pythonpath: False
    opencascade: False
    tinyasm: False
    torch: False
    petsc_int_type: int32
    cache_dir: /home/alex/firedrake/.cache
    complex: False
    remove_build_files: False
    with_blas: None
    netgen: False
Additions:
    None
Environment:
    PYTHONPATH: None
    PETSC_ARCH: None
    PETSC_DIR: None
 

Status of components:
---------------------------------------------------------------------------
|Package             |Branch                        |Revision  |Modified  |
---------------------------------------------------------------------------
|FInAT               |master                        |e2805c4   |False     |
|PyOP2               |master                        |fbde61f9  |False     |
|fiat                |master                        |e7b2909   |False     |
|firedrake           |master                        |c5e939dde |False     |
|h5py                |firedrake                     |4c01efa9  |False     |
|libspatialindex     |master                        |4768bf3   |True      |
|libsupermesh        |master                        |dbe226b   |False     |
|loopy               |main                          |8158afdb  |False     |
|petsc               |firedrake                     |09f36907a6|False     |
|pyadjoint           |master                        |f194553   |False     |
|pytest-mpi          |main                          |a478bc8   |False     |
|tsfc                |master                        |799191d   |False     |
|ufl                 |master                        |054b0617  |False     |
---------------------------------------------------------------------------
```

**Comment by Alexey-Voronin** [comment]
This PR fixes the issue, but it would be great if someone who knows about #3414 could take a look.

**Comment by connorjward** [comment]
Fixed by #3414 (merged)
